pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -628,6 +628,19 @@ func (r *ReconcileInfinispan) scheduleUpgradeIfNeeded(infinispan *infinispanv1.I
 	// Get Infinispan image that the operator creates
 	desiredImage := consts.DefaultImageName
 
+	switch infinispan.Spec.Upgrade.Policy {
+	case infinispanv1.UpgradePolicyManual:
+		// In manual mode upgrade is triggered by the user changing infinispan.spec.Image field
+		// if that field is empty the operator default image is used (same as policy latest in this case)
+		if *infinispan.Spec.Image != "" {
+			desiredImage = *infinispan.Spec.Image
+		}
+	case infinispanv1.UpgradePolicyCVE: // All the rest upgrade mode works as usual atm
```

Comment:
I think we should only provide the "Manual" and "latest" strategies until we're able to implement this properly, otherwise it will cause confusion with users.